### PR TITLE
[Bugfix] Build artifact download

### PR DESCRIPTION
**Pull request by kloessst**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)

### Motivation and Context

See issue #2628. The download fails because basic authentication (name+password) is required, but we try to authenticate with a token.

### Description

- Use basic authentication for artifact download by adding a case to BambooAuthorizationInterceptor
- Display real filename with extension for download

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise
4. Navigate to the template build plan
image
5. Click on _Configure build plan_
image
6. Select the default job
image
7. Alter the goal of the maven task as seen in the screenshot
image
8. In the same view click on tab _Artifacts_
9. Create an artifact with the following settings
image
10. Participate in the exercise and solve all test cases (copy the solution)
11. A download build result button should be shown. Click on it.
image
--> Artemis should provide a download for the jar and display the fileName with the .jar extension

### Test Coverage

participation.service: 41% -> 45%

### CODE REVIEW

File: src/test/javascript/spec/service/participation.service.spec.ts [hunk 1]
```diff
@@ -102,6 +103,26 @@ describe('Participation Service', () => {
...
     });
 
+    it.each<any>([
+        ['attachment; filename="FixArtifactDownload-Tests-1.0.jar"', 'FixArtifactDownload-Tests-1.0.jar'],
+        ['', 'artifact'],
+        ['filename="FixArtifactDownload-Tests-1.0.jar"', 'FixArtifactDownload-Tests-1.0.jar'],
+        ['f="abc"', 'artifact'],
+    ])('%# should download artifact and extract file name: %p', async (headerVal: string, expectedFileName: string, done: jest.DoneCallback) => {
+        const expectedBlob = new Blob(['abc', 'cfe'], { type: 'application/java-archive' });
+        let headers = new HttpHeaders({ 'content-disposition': headerVal, 'content-type': 'application/java-archive' });
```
Comment (krusche) [inline src/test/javascript/spec/service/participation.service.spec.ts:113]:
```suggestion
        const headers = new HttpHeaders({ 'content-disposition': headerVal, 'content-type': 'application/java-archive' });
```

**Review by krusche — APPROVED** [review]
except the small GitHub issue, the code looks good 👍 thanks for this nice contribution

**Review by EneaGore — COMMENTED** [review]
I get "Access denied. Sorry, you have insufficient permissions to view the page." on ts1 when i try to Access the Template Build plan, but i can view the Template Build Plans from the Live Server...
Edit: All good, wrong account on Bamboo

**Comment by kloessst** [comment]
> 
> 
> I get "Access denied. Sorry, you have insufficient permissions to view the page." on ts1 when i try to Access the Template Build plan, but i can view the Template Build Plans from the Live Server...

Are you instructor/tutor in the course?

**Review by EneaGore — APPROVED** [review]
Everything works as described for me on ts1.

**Review by airelawaleria — APPROVED** [review]
Tested on ts1. Works fine

**Review by simonzilker — APPROVED** [review]
Everything worked perfectly for me. Thanks!

**Review by marvinchimeka — APPROVED** [review]
No issues found. Approve. Tested on ts 1

**Review by tiffca — APPROVED** [review]
Tested successfully

**Review by JohannesWeiss — APPROVED** [review]
Approve as well.

**Review by FlorianStang — APPROVED** [review]
Tested on TS1 everything worked as intended.

**Review by krusche — APPROVED** [review]

**Pull request merged**